### ISSUE TITLE
[Bug]: Sorting Icon in DataTable does not show visible focus on hover

### ISSUE THREAD

**Issue by mahalakshminadigatla** [carbon-design-system/carbon#20874]
### Package

@carbon/react

### Package version

1.83.0

### Description

The sorting icon (up/down arrow) beside the header text does not display a visible focus indicator when hovered or focused.
This violates [Accessibility Requirement 1.4.13 – Content on Hover or Focus,](https://www.w3.org/WAI/WCAG22/UNDERSTANDING/content-on-hover-or-focus.html) which requires that interactive elements provide a clear visual indication when they receive pointer hover or keyboard focus.

### Reproduction/example

https://stackblitz.com/edit/github-vpbxjtu2-jzzsxkvm?file=src%2FApp.jsx

### Steps to reproduce

when you keep the curser on the sorting icon, the icon beside the text in the header cell does not display a visible focus indicator for sorting icon alone. 

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by 2nikhiltom** [comment]
Hey @mahalakshminadigatla - I think this is expected behaviour , the sort icons/svg are part of a button when receive focus when navigated via keyboard/mouse. 

Do you see any A11Y violation thrown by accessibility checker ? 

<img width="254" height="151" alt="Image" src="https://github.com/user-attachments/assets/bfa3b50c-6625-43fc-8a15-0f6b4751ca6c" />

<img width="1307" height="411" alt="Image" src="https://github.com/user-attachments/assets/675a9c56-8a52-4756-8918-e0302e0deb79" />

**Comment by heloiselui** [comment]
Hey @mahalakshminadigatla, I'm going to close this issue since this is expected behaviour, but if you spot any violations we missed, we can reopen it. Thank you!!
